src/dune_pkg/fetch.mli

```diff
@@ -19,6 +19,31 @@ val fetch
   -> OpamUrl.t
   -> (unit, failure) result Fiber.t
 
+module Git_repo : sig
```

Comment:
Do you think you could move this to a separate library and add some unit tests?

Reply:
Will do! But first I'd like to rebase this upon #8962, then I can squash all the intermediate mess.

Reply:
I've added some expect tests. Do you think unit tests should also be added?

Reply:
I don't see any expect tests, do you mean cram tests?

In any case, yes I think that unit tests (in the form expect tests) should be added. It doesn't have to be in this PR though.